src/main/java/com/getindata/connectors/http/internal/sink/httpclient/BatchRequestSubmitter.java

```diff
@@ -0,0 +1,146 @@
+package com.getindata.connectors.http.internal.sink.httpclient;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpClient.Version;
+import java.net.http.HttpRequest.BodyPublisher;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.annotation.VisibleForTesting;
+
+import com.getindata.connectors.http.internal.config.HttpConnectorConfigConstants;
+import com.getindata.connectors.http.internal.sink.HttpSinkRequestEntry;
+
+@Slf4j
+public class BatchRequestSubmitter extends AbstractRequestSubmitter {
```

Comment:
New thing - core logic for submitting requests as Json array in batch in one HTTP query.